transport/http/client.go

```diff
@@ -85,6 +101,10 @@ func (c Client) Endpoint() endpoint.Endpoint {
 		}
 
 		resp, err := ctxhttp.Do(ctx, c.client, req)
+		for _, f := range c.after {
+			ctx = f(ctx, resp)
+		}
+
```

Comment:
My bad, you are correct, this should be after the error check. I looked into `ctxhttp.Do` and since no response is being passed back, this doesn't make sense. This now appears after the error, but before decoding, which is nice because it more closely mirrors `Server.ServeHTTP`. Comments have also been updated to reflect this change. Thanks!